src/main/kotlin/org/rust/lang/core/psi/RustNumber.kt

```diff
@@ -0,0 +1,79 @@
+package org.rust.lang.core.psi
+
+import com.intellij.psi.tree.IElementType
+import org.rust.lang.core.psi.RustNumber.Float.F32
+import org.rust.lang.core.psi.RustNumber.Float.F64
+import org.rust.lang.core.psi.RustNumber.Int.*
+import org.rust.lang.core.psi.RustTokenElementTypes.FLOAT_LITERAL
+import org.rust.lang.core.psi.RustTokenElementTypes.INTEGER_LITERAL
+import java.lang.Long.parseLong
+
+sealed class RustNumber<out T>(val valueString: String) {
```

Comment:
I think that this `RustNumber` is a bit over engineered. The only real usage of it is https://github.com/intellij-rust/intellij-rust/pull/263/files#diff-dbc28844fa6b82b3ec73c98ba019e118R25. Also I am worried about how this will work with type inference, when we have one. I don't think that we should distinguish between various types of integers right now.